{openresty,nginxModules.brotli}: updates and improvements

emilazy opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

This updates the `ngx_brotli` module to the latest upstream version, and also rebases the `openresty` package on top of the `nginx/generic.nix` infrastructure, simplifying things considerably and allowing the use of third-party modules with OpenResty.

I haven't tested this extensively yet, but the build completes successfully for me and the OpenResty codebase is just a distribution of nginx, so I think it should be pretty low-risk. `nix build '(with (import ./. {}); openresty.override { modules = with nginxModules; [ brotli ]; })'` builds successfully with these changes.

cc @thoughtpolice does this look reasonable to you?

@GrahamcOfBorg build nginx openresty

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

emilazy committed:
nginxModules.brotli: v0.1.2 -> unstable

The fork was merged back upstream but has yet to see a formal release.

emilazy commented:
It turns out that naming a parameter `src` causes exciting breakage when you `callPackage` it (because `pkgs.src` exists); I broke the nginx build.

Let's try that again...

@GrahamcOfBorg build nginx openresty

thoughtpolice commented:
Sorry for the delay reviewing this @emilazy -- yes, these changes look pretty good! Please rebase (it looks like there's just been a trivial update to `nginxMainline`) and I'll merge, sorry for the delay here.

emilazy committed:
openresty: rebase on top of nginx package

The primary motivation of this change was to allow third-party modules
to be used with OpenResty, but it also results in a significant
reduction of code duplication.

emilazy commented:
should be good now :)

@GrahamcOfBorg build nginx openresty
